node/subsystem-util/src/rolling_session_window.rs

```diff
@@ -146,19 +158,103 @@ impl RollingSessionWindow {
 		self.earliest_session + (self.session_info.len() as SessionIndex).saturating_sub(1)
 	}
 
+	async fn earliest_non_finalized_block_session<Sender>(
+		&self,
+		sender: &mut Sender,
+	) -> Result<u32, SessionsUnavailable>
+	where
+		Sender: overseer::SubsystemSender<RuntimeApiMessage>
+			+ overseer::SubsystemSender<ChainApiMessage>,
+	{
+		let last_finalized_height = {
+			let (tx, rx) = oneshot::channel();
+			sender.send_message(ChainApiMessage::FinalizedBlockNumber(tx)).await;
+			match rx.await {
+				Ok(Ok(number)) => number,
+				Ok(Err(e)) =>
+					return Err(SessionsUnavailable {
+						kind: SessionsUnavailableReason::ChainApi(e),
+						info: None,
+					}),
+				Err(err) => {
+					gum::warn!(
+						target: LOG_TARGET,
+						?err,
+						"Failed fetching last finalized block number"
+					);
+					return Err(SessionsUnavailable {
+						kind: SessionsUnavailableReason::MissingLastFinalizedBlock,
+						info: None,
+					})
+				},
+			}
+		};
+
+		let (tx, rx) = oneshot::channel();
+		// We want to get the parent of the last finalized block.
```

Comment:
I don't see where we get a parent here. It seems, that we extract the block hash at `last_finalized_height`, so either the comment (most likely) or the logic is wrong. For the parent of the last finalized block, we need to get a hash for `last_finalized_height - 1` I suppose, but from the following code, it seems that we actually need the `last_finalized_height` block hash.

Reply:
Yeah, I have changed the code but forgot to do the same about then comment. The last finalized child session is what we are after here.